### PR TITLE
Ammo rebalance, new antag ammo, sprites

### PR DESCRIPTION
**Pull request by Voldirs**
1)Firearm damage rebalance
Basic idea was to separate low damage smg/pistol rounds, medium damage rifle rounds, high damage revolver/rare pistol rounds and to exclude some strange things like .357 being more powerful than .44 magnum.
This also includes rubber bullets changes, they got less brute damage in general and some pain damage rebalance. 12g beanbag is a beast now

2)New ammo type
I replace old ammo with "high-velocity" variant in uplink. So now you have 2 choices of getting ammo as antag with uplink. Either you can find/buy common ammo or diskette, either you can spend some telecrystals in exchange for much more punch and armor piercing capabilities in a shot.

3)Colored stripes on rubber or high velocity mags
Blue stripe is for rubber ammo, Red one is for high-velocity ammo.
scrnshot1

### CODE REVIEW

**Comment by NanakoAC** [comment]
I did a bit of planning, here's some thoughts on an overview for gun stats. 
The AKMS and Sol would both qualify as light rifles here

Where nonlethal ammo is concerned, the damage should remain the same total, but most of it should be agony rather than brute

The ranges specified here give some room to manoeuvre in order to balance individual weapons up or down based on rarity, reputation, player expectations, etc.

The damage values of bullet types can be adjusted for a baseline, and damage multipliers set on individual guns to farther vary them

Gun categories:

Pistol (light)
	Small size
	10-25 damage per shot
	No armor penetration
	
Pistol (heavy), (includes revolvers)
	Small or normal size
	20-35 damage per shot
	5-15 armor penetration
	
SMG:
	Normal or Large size
	5-15 damage per shot
	Full auto firing mode
	No armor penetration
	
Automatic Rifle (light):
	Large or huge size
	10-20 damage per shot
	5-10 armor penetration
	may or may not have full auto fire
	
Automatic Rifle (Heavy)
	Large or huge size
	15-30 damage per shot
	Full auto fire
	10-15 armor pentration
	
Light Machine Gun:
	Huge Size
	15-30 damage per shot
	10-15 armor penetration
	Full auto fire
	Huge magazine capacities (80 rounds usually)
	Higher firing rate (600 RPM, rather than 400 that most guns get)
	
Shotgun:
	30-70 peak damage per shot (assuming most/all pellets hit)
	No armor penetration
	
Sniper Rifles
	Huge Size
	40-80 damage per shot
	30 armor penetration
	Scoped fire
	Probably can fire through walls
	Limited mag capacity (often just one round)

**Comment by Voldirs** [comment]
Automatic rifles and SMGs damage is slightly reduced.
Reduced damage for bull shotgun because it's too hard punching with not so hard reload.
Reduced damage for sawnoff shotgun because it's compact
Increased damage for olivaw to make it decent pistol. It's cost is also higher.

**Comment by Wesreidau** [comment]
Nanako, I have thoughts on your planning.

I will also give a list of ammo categories by my estimation along with common weapons you might be familiar with. There appears to be 25 or so ammo types presently. I think we can halve that and consolidate a lot of rarely used ammunition types, and create a general pattern based on real-world cartridge niches.

I disagree with the general notion agony is bad for gameplay. Agony is what lets you take a hit, be pulled to safety by a friend and get up. It is also what makes you stop dumping your magazine into people as you take hits.

In this, 'Agony' is used to represent stopping power. Some projectiles, like low velocity, high bore pistols, have a lot of stopping power due to the mass of the projectile and the transfer of energy as it comes to a full stop in the target. Others, like high-power rifle rounds, tend to pass entirely through the target and not transfer as much energy, even though they may cause worse actual injuries. These are more lethal to the target, but may not drop him outright.

There is also the chance of a bullet embedding in the target, or that bullet becoming shrapnel. Many projectiles remain whole, however certain types, such as high velocity, small bore rifle rounds, break up into an explosion of fragments. This sort of injury throws the normal progression off. A 5.56 round which fragments into an explosion of lead bits creates a far worse wound channel than the 7.62 round that passes out the back of the target. However, NATO insisted on their idiot steel-cored M855 cartridge from the damn Belgians instead of God-fearing M193, so now the M4 shoots inaccurate off-center steel core ice picks. For this reason I included an armored-piercing version.

High Velocity, Small Bore Pistol (9mm)
Revolvers, service pistols, MP-5.
Lower damage, better armor penetration.
More cartridges per magazine (usually)
Lower Agony
Rubber version.

Low Velocity, Large Bore Pistol (.45 ACP)
Revolvers, 1911, UMP-45.
Higher damage, less armor penetration.
Less cartridges per magazine (usually)
High Agony
Rubber version.

Hand Cannon Pistol (Magnum)
High power revolvers, Desert Eagle
High damage, high armor penetration
Less cartridges per magazine, larger pistols
High agony

High Velocity, Small Bore Rifle, Ball ( 5.56mm)
M-16.
Medium armor piercing, medium damage.
30 round magazines, smaller sized.
Medium Agony, high chance of imbedded shrapnel (bullets break up)

High Velocity, Small Bore Rifle, AP ( 5.56mm)
M-16.
High armor piercing, low damage.
30 round magazines, smaller sized.
Low agony 

Mid Velocity, Mid Bore Rifle, Ball ( 6.5mm)
Sol, L6 SAW
Medium armor piercing, medium damage.
30-round magazines.
Medium agony.

Low Velocity, Large Bore Rifle, Ball (7.62)
AK-47.
Medium armor piercing, medium damage.
30-round magazines, larger sized.
High agony.

High Velocity, Large Bore Rifle (7.62x54, .30)   Currently 7.62mm.
Battle Rifles, LMGs, Sniper Rifles
High armor piercing, high damage.
20-round box, 100-round belts, 5-round clips
High agony.

Shoulder Cannon (14.5, .50 BMG)
AMRs, hardsuit-mounted heavy weapons.
Very high AP, high damage.
Single shot.
Extreme agony.

This goes from 25 ammo types to 11 ammo types. A few flavor types for factional flavor (a standard pistol and rifle cartridge for Zhi Xi Ying, for example), but we should eliminate things like .357 magnum and .44 magnum and .38 special in favor of rolling these into a hypothetical magnum cartridge like 12.5mm.

I also think we should cull rubber bullets from all rifles, as these weapons have very high base damage values. I also worry people will full-auto their rubber bullet machine guns. For the submachine guns this is less dangerous, but the rifle rounds will be very bad for everyone.

Thus we see...

9mm HVP (Civilian pistols, revolvers, military/police SMGs)
9mm HVP Rubber
11mm LVP (Civilian pistols, revolvers, military/police SMGs)
11mm LVP Rubber
12.5mm Magnum (Specialty Pistols)

5.56mm HVR (Not-Frozen-Star assault rifle)
6.5mm MVR  (Frozen Star assault rifle)
7.62mm LVR (Soviet Meme assault rifle)

8mm HPR (Not-Frozen-Star full-power rifle)
(Frozen Star's 6.5mm is intended as universal)
7.62x54R HPR (Soviet Meme full-power rifle)


14.5 AMR (Anti-material rifle)



And of course, shotgun shells.

Slug
High brute damage.
Medium armor piercing.
Medium agony.

Beanbag (needs an accuracy nerf, IRL beanbag rounds are very inaccurate)
No armor piercing value.
High agony, deals some brute damage.

Shot
Higher potential damage output than slugs.
Lowest armor piercing value.
Highest agony potential.

Shot, Rubber (multiple rubber pellets to get everyone to shut up and sit down)
Low agony, No brute damage.
Medium agony potential.

Flash (nobody uses pistol flashes)
Brute damage when impacting a mob.
Very small flashbang effect.

Chemical (pepper)
Brute damage when impacting a mob.
One chemical cloud on impact, typically pepper.

File: code/modules/projectiles/projectile/bullettypes.dm [hunk 1]
```diff
@@ -82,13 +101,19 @@
 
 /obj/item/projectile/bullet/a762
 	damage = 20
-	armor_penetration = 15
+	armor_penetration = 10
 	penetrating = 1
 	can_ricochet = TRUE
 
+/obj/item/projectile/bullet/a762h
```
Comment (NanakoAC) [inline code/modules/projectiles/projectile/bullettypes.dm]: For these high velocity rounds, we should establish a standard

I'm thinking 20% higher, in both damage and AP
So give these ones 24 damage and 18 AP

this leaves a possibility open for hollowpoints and AP rounds to be added later which specialise in only one of these attributes

**Comment by Voldirs** [comment]
I like that idea about agony addition, but I'm not going to fumble with that in this PR. Overall idea is good. I agree on that standartization, let's give it a try.

**Comment by NanakoAC** [comment]
you may need to cherrypick your code into a new branch and make a new PR, it seems to think you've made map edits here

**Comment by NanakoAC** [comment]
also the conflict with ammo.dmi is my fault, i added special ammobox sprites for the PK

you can copy those into your version and overwrite

**Comment by shazbot194** [comment]
What if instead of just adding in HV ammo, we add in dum dum rounds which are red, AP which can be green. Dum Dum rounds would do more damage but have less AP, and AP would do less damage, but have a lot more AP.

**Comment by NanakoAC** [comment]
thats already planned shaz, seperate thing.

**Comment by shazbot194** [comment]
My apologies

**Comment by Voldirs** [comment]
I'll make fresh branch and PR

**Pull request closed without merging**